gremlin-driver/src/main/java/org/apache/tinkerpop/gremlin/driver/Client.java

```diff
@@ -561,16 +568,56 @@ protected void initializeImplementation() {
 
         private Consumer<Host> initializeConnectionSetupForHost = host -> {
             try {
-                // hosts that don't initialize connection pools will come up as a dead host
+                // hosts that don't initialize connection pools will come up as a dead host.
                 hostConnectionPools.put(host, new ConnectionPool(host, ClusteredClient.this));
 
-                // added a new host to the cluster so let the load-balancer know
+                // hosts are not marked as available at cluster initialization, and are made available here instead.
+                host.makeAvailable();
+
+                // added a new host to the cluster so let the load-balancer know.
                 ClusteredClient.this.cluster.loadBalancingStrategy().onNew(host);
             } catch (RuntimeException ex) {
-                final String errMsg = "Could not initialize client for " + host;
-                throw new RuntimeException(errMsg, ex);
+                throw new RuntimeException(String.format("Could not initialize client for %s.", host), ex);
             }
         };
+
+        private void handleUnavailableHosts(List<Host> unavailableHosts, ExecutorService hostExecutor) {
```

Comment:
why do we need the hostExecutor to re-initialize the host? Asking because makeUnavailable() already uses the cluster's scheduler pool asynchronously to run the lambda. We can hand over the reconnect/re-initialize responsibility to the scheduler from here.

Reply:
Yes, I missed that part. Changed it. Thanks!